src/discovery/Discovery.ts

```diff
@@ -17,22 +16,28 @@ import type Sigchain from '../sigchain/Sigchain';
 import type KeyManager from '../keys/KeyManager';
 import type { ClaimIdEncoded, Claim, ClaimLinkIdentity } from '../claims/types';
 import type { ChainData } from '../sigchain/types';
+import type TaskManager from '../tasks/TaskManager';
+import type { ContextTimed } from '../contexts/types';
+import type { PromiseCancellable } from '@matrixai/async-cancellable';
+import type { TaskHandler, TaskHandlerId } from '../tasks/types';
 import Logger from '@matrixai/logger';
 import {
   CreateDestroyStartStop,
   ready,
-  status,
 } from '@matrixai/async-init/dist/CreateDestroyStartStop';
-import { IdInternal } from '@matrixai/id';
-import * as idUtils from '@matrixai/id/dist/utils';
-import * as discoveryUtils from './utils';
 import * as discoveryErrors from './errors';
+import * as tasksErrors from '../tasks/errors';
 import * as nodesErrors from '../nodes/errors';
 import * as networkErrors from '../network/errors';
 import * as gestaltsUtils from '../gestalts/utils';
 import * as claimsUtils from '../claims/utils';
 import * as nodesUtils from '../nodes/utils';
-import { never, promise } from '../utils';
+import { never } from '../utils';
+import { context } from '../contexts/index';
+import TimedCancellable from '../contexts/decorators/timedCancellable';
+
+const abortSingletonTaskReason = Symbol('abort singleton task reason');
```

Comment:
Provide doc block comment about what this reason means.

Reply:
Added comments.